apps/newsletters-ui/src/app/components/StepNav.tsx

```diff
@@ -0,0 +1,33 @@
+import { Step, StepLabel, Stepper } from '@mui/material';
+import type { StepListing } from '@newsletters-nx/state-machine';
+
+interface Props {
+	currentStepId?: string;
+	stepList: StepListing[];
+	onEditTrack: boolean;
+}
+
+export const StepNav = ({ currentStepId, stepList, onEditTrack }: Props) => {
+	const filteredStepList = stepList.filter((step) => {
+		switch (step.role) {
```

Comment:
In my IDE I'm getting "Unsafe. member access .role on an 'any' value" (ditto id and label on lines 27 and 28)